From 2fed1fc9dae6d84da628f059d43da938b644f15c Mon Sep 17 00:00:00 2001
From: Houssem Dafdouf <houssem.dafdouf_ext@softathome.com>
Date: Fri, 23 May 2025 10:34:41 +0200
Subject: [PATCH] gcc: fix build errors when using gcc v13.3.0

When building prplMesh using recent gcc verion like v13.3.0 some build
errors are occuring :

- Unkown uint32_t, int64_t types

-> Solution : include <cstdint> header

- Calls like std::move(std::string("agent")); (from
platform_manager.cpp and backhaul_manager.cpp are raising a warning
(with -Werror enabled)

-> Solution: change this to std::string("agent");

Closes: PPW-569

Signed-off-by: Houssem Dafdouf <houssem.dafdouf_ext@softathome.com>
---
 agent/src/beerocks/slave/backhaul_manager/backhaul_manager.cpp  | 2 +-
 agent/src/beerocks/slave/platform_manager/platform_manager.cpp  | 2 +-
 .../beerocks/bcl/include/bcl/network/netlink_event_listener.h   | 1 +
 common/beerocks/bwl/include/bwl/key_value_parser.h              | 1 +
 4 files changed, 4 insertions(+), 2 deletions(-)

diff --git a/agent/src/beerocks/slave/backhaul_manager/backhaul_manager.cpp b/agent/src/beerocks/slave/backhaul_manager/backhaul_manager.cpp
index 2e640322b2..15e1f71308 100644
--- a/agent/src/beerocks/slave/backhaul_manager/backhaul_manager.cpp
+++ b/agent/src/beerocks/slave/backhaul_manager/backhaul_manager.cpp
@@ -1544,7 +1544,7 @@ bool BackhaulManager::handle_slave_backhaul_message(int fd, ieee1905_1::CmduMess
     case beerocks_message::ACTION_BACKHAUL_REGISTER_REQUEST: {
 
         m_agent_fd      = fd;
-        auto agent_name = std::move(std::string("agent"));
+        auto agent_name = std::string("agent");
         LOG(DEBUG) << "Assigning FD (" << fd << ") to " << agent_name;
         m_cmdu_server->set_client_name(fd, agent_name);
 
diff --git a/agent/src/beerocks/slave/platform_manager/platform_manager.cpp b/agent/src/beerocks/slave/platform_manager/platform_manager.cpp
index 42c5db2803..6ffd8f6371 100644
--- a/agent/src/beerocks/slave/platform_manager/platform_manager.cpp
+++ b/agent/src/beerocks/slave/platform_manager/platform_manager.cpp
@@ -414,7 +414,7 @@ bool PlatformManager::handle_cmdu(int fd, ieee1905_1::CmduMessageRx &cmdu_rx)
             return false;
         }
 
-        auto agent_name = std::move(std::string("agent socket"));
+        auto agent_name = std::string("agent socket");
         LOG(DEBUG) << "Assigning FD (" << fd << ") to " << agent_name;
         m_cmdu_server->set_client_name(fd, agent_name);
 
diff --git a/common/beerocks/bcl/include/bcl/network/netlink_event_listener.h b/common/beerocks/bcl/include/bcl/network/netlink_event_listener.h
index f8174e4bde..24f50ad7b6 100644
--- a/common/beerocks/bcl/include/bcl/network/netlink_event_listener.h
+++ b/common/beerocks/bcl/include/bcl/network/netlink_event_listener.h
@@ -11,6 +11,7 @@
 
 #include <linux/netlink.h>
 
+#include <cstdint>
 #include <functional>
 #include <unordered_map>
 
diff --git a/common/beerocks/bwl/include/bwl/key_value_parser.h b/common/beerocks/bwl/include/bwl/key_value_parser.h
index f5cd993f15..948980538c 100644
--- a/common/beerocks/bwl/include/bwl/key_value_parser.h
+++ b/common/beerocks/bwl/include/bwl/key_value_parser.h
@@ -9,6 +9,7 @@
 #ifndef _BWL_KEY_VALUE_PARSER_H_
 #define _BWL_KEY_VALUE_PARSER_H_
 
+#include <cstdint>
 #include <list>
 #include <string>
 #include <unordered_map>
-- 
GitLab

